src/spline.rs

```diff
@@ -80,7 +83,7 @@ impl<T, V> Spline<T, V> {
 
   /// Retrieve the keys of a spline.
   pub fn keys(&self) -> &[Key<T, V>] {
-    &self.0
+    self.0.as_slices().0
```

Comment:
Assuming we always push to the back of the slice, the first element of the tuple returned by `as_slices` should correctly contain all the keys.
Therefor, it's important to keep in mind to always push keys in a consistent way.

Reply:
> Assuming we always push to the back of the slice, the first element of the tuple returned by as_slices should correctly contain all the keys.

I agree with this, but now I’m starting to think that the API is a bit weird; we can remove and update keys at random position but we can only add at the end. Within the context of your PR, this is fine, but I think we should add a small comment about that here.

Reply:
I think that removing at random positions is fine since it doesn't alter the sort of the collection.
Adding is different, since it doesn't make sense to add at an index specified by the user: the collection will be sorted again anyway, losing the correspondence with the index.

Reply:
The `add` method should only push to the back so that we can minimise the number of needed `internal_sort` calls.

Reply:
Agreed.